apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/trace/AbstractTracingSpan.java

```diff
@@ -301,6 +302,13 @@ public void ref(TraceSegmentRef ref) {
         if (refs == null) {
             refs = new LinkedList<>();
         }
+        /*
+         * There maybe an exception in some plugins, which causes the span to fail to exit,
+         * which results in the ThreadLocal not being cleaned up, and the refs data is increasing.
```

Comment:
```suggestion
         * Provide the OOM protection if the entry span hosts too many references.
```
You should not do coding for bugs. This OOM protection could happen w/o bug, such as consuming 1k+ message as a batch, isn't it?

Reply:
done.